File: src/main/java/com/kuroko/properties/MiraiProperties.java
package com.kuroko.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "mirai")
@Data
public class MiraiProperties {

    private String protocol;
    private Long account;
    private String password;
    private boolean disableContactCache;

}

File: src/main/kotlin/com/hkuroko/mirai/MiraiProperties.kt
package com.hkuroko.mirai

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "mirai")
data class MiraiProperties(
    val protocol: String = "ANDROID_PHONE",
    val qq: Long,
    val pwd: String,
    val deviceJson: String,
    val disableContactCache: Boolean = false,
    val fileCacheStrategy: String = "TEMP",
    val cacheDir: String = "cache",
    val autoReconnectOnForceOffline: Boolean = false,
    val reconnectionRetryTimes: Int = 5,
    val highwayUploadCoroutineCount: Int = 2,
    val autoLogin: Boolean = true
)